pkg/api/gitops/v1alpha1/application.go

```diff
@@ -305,7 +305,13 @@ type Application struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec ApplicationSpec `json:"spec,omitempty"`
+	Spec   ApplicationSpec   `json:"spec,omitempty"`
+	Status ApplicationStatus `json:"status,omitempty"`
```

Comment:
How about adding the status fields as subresource? Please refer to https://book.kubebuilder.io/reference/generating-crd.html?highlight=//+kubebuilder:subresource:status#subresources

Reply:
BTW, this won't block this PR from merging by other reviewers.